### PR TITLE
Added support for extra file extensions

### PR DESCRIPTION
**Pull request by stefounet**
Hi Brent

Here is the extra ext option we discussed.

### CODE REVIEW

File: HELP [hunk 1]
```diff
@@ -7,3 +7,4 @@ Options:
    --reporter           custom reporter
    --jslint-reporter    use a jslint compatible xml reporter
    --show-non-errors    show additional data generated by jshint
+   --extra-ext          comma-separated list of file extension to use (.js is default)
```
Comment (brentlintner) [inline HELP]: LOL... This seems so minute to bring up (and cause a rebase..)... but, perhaps `extension` should be pluralized?

File: README.md [hunk 2]
```diff
@@ -48,6 +48,12 @@ Specify custom lint options (see [example/config.json](https://github.com/jshint
 
 Note: This bypasses any .jshintrc files.
 
+## File Extensions
+
+Default extension for files is ".js". If you want to use JSHint with other file extensions (.json), you need to pass this extra extension as an option
```
Comment (brentlintner) [inline README.md]: Perhaps a `:` or `.` at the end of this sentence? (not a big deal, but thought I would mention, for consistency)

**Comment by niknah** [comment]
When you type "jshint xxx.txt", it should error instead of doing nothing and failing silently.  

Took me a while to figure out that it only took .js

File: HELP (hunk 1 above)
Comment (stefounet) [inline HELP]: yep, sure !

File: README.md (hunk 2 above)
Comment (stefounet) [inline README.md]: idem

**Comment by stefounet** [comment]
Yep, you're right ! It also took me a while to find out why my json file wasn't parsed ...
We should add a message to show ignored files during the collect phase.

**Comment by brentlintner** [comment]
@stefounet @niknah Cool, thanks for that. I created a separate issue for that. https://github.com/jshint/node-jshint/issues/97

**Comment by brentlintner** [comment]
Also, there appear to be some lint errors (via the `jake lint` task) in this branch.

**Comment by smagch** [comment]
I hope jshint doesn't sniff file extensions. I'm personally coding cli which has no extension and start from `#!/usr/bin/env node`. I'm ending up with

``` javascript
mv ./bin/jscat ./bin/jscat.js
jshint ./bin/jscat.js
mv ./bin/jscat.js ./bin/jscat
```

**Comment by brentlintner** [comment]
@smagch It has been a pain point, but what other way to identify the file (cross platform) without picking up all files? `#!` is not 100% reliable (IMO), but may be a band aid solution.. however it does not (fully) solve the problem (i.e. it won't match a file like [this](https://github.com/jshint/node-jshint/blob/master/Jakefile)). There is an open issue [here](https://github.com/jshint/node-jshint/issues/98) for that, open to suggestions! :-)

**Comment by brentlintner** [comment]
Thanks again. Published v0.6.2 with this (after rebasing, fixing some conflicts, and re-testing). I also squashed the commits into one (for succinctness). Hopefully that was ok. Let me know if there are any discrepancies!

https://github.com/jshint/node-jshint/commit/86f618b77212dae1777539c9f72052d414121520

**Comment by stefounet** [comment]
Ok, nice !
It seems that everything is ok !

Le 30/04/12 17:47, Brent Lintner a écrit :

> Thanks again. Published v0.6.2 with this (after rebasing, fixing some conflicts, and re-testing). I also squashed the commits into one (for succinctness). Hopefully that was ok. Let me know if there are any discrepancies!
> 
> https://github.com/jshint/node-jshint/commit/86f618b77212dae1777539c9f72052d414121520
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/jshint/node-jshint/pull/96#issuecomment-5420262

## 

Stéphane RIOS - fasterize.com - 06 30 26 44 80

**Pull request closed without merging**